clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/model/EcsInstance.java

```diff
@@ -0,0 +1,82 @@
+package com.netflix.spinnaker.clouddriver.ecs.model;
+
+import com.amazonaws.services.ec2.model.InstanceStatus;
+import com.amazonaws.services.ecs.model.Task;
+import com.netflix.spinnaker.clouddriver.ecs.EcsCloudProvider;
+import com.netflix.spinnaker.clouddriver.model.HealthState;
+import com.netflix.spinnaker.clouddriver.model.Instance;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+public class EcsInstance implements Instance, Serializable {
+  private String name;
+  private HealthState healthState;
+  private Long launchTime;
+  private String zone;
+  private List<Map<String, String>> health;
+  private String providerType;
+  private String cloudProvider;
+
+  public EcsInstance(String name, Task task, InstanceStatus ec2Instance) {
+    this.name = name;
+    providerType = cloudProvider = EcsCloudProvider.ID;
+    launchTime = task.getStartedAt().getTime();
+    health =  null;
+    healthState = calculateHealthState(task.getLastStatus(), task.getDesiredStatus());
+    zone = ec2Instance.getAvailabilityZone();
+  }
+
+  @Override
+  public String getZone() {
+    return zone;
+  }
+
+  @Override
+  public List<Map<String, String>> getHealth() {
+    return health;
```

Comment:
the payload for the Instance endpoint for EC2 instances is quite interesting.  Here is what the "health" key returns:

```  "health": [
    {
      "healthClass": "platform",
      "state": "Unknown",
      "type": "Amazon"
    },
    {
      "instanceId": "i-0b9bca7615be40e14",
      "loadBalancers": [
        {
          "description": "N\/A",
          "healthState": "Up",
          "instanceId": "i-0b9bca7615be40e14",
          "loadBalancerName": "armoryspinnaker-ha-internal",
          "loadBalancerType": "classic",
          "reasonCode": "N\/A",
          "state": "InService"
        }
      ],
      "state": "Up",
      "type": "LoadBalancer"
    }
  ],```

I don't think we should do that much at this point though, for the POC.  What you've covered is good enough for POC

Reply:
Correct. This will be very ECS specific, with values that we find important as we develop this story